cmd/apprepository-controller/pkg/apis/apprepository/v1alpha1/types.go

```diff
@@ -35,9 +36,15 @@ type AppRepository struct {
 
 // AppRepositorySpec is the spec for an AppRepository resource
 type AppRepositorySpec struct {
-	Type           string `json:"type"`
-	URL            string `json:"url"`
-	ResyncRequests uint   `json:"resyncRequests"`
+	Type           string             `json:"type"`
+	URL            string             `json:"url"`
+	Auth           *AppRepositoryAuth `json:"auth,omitempty"`
```

Comment:
I think if you don't make this a pointer it will always be initialised to an empty `AppRepositoryAuth` which means you only need to check `apprepo.Spec.Auth.Header != nil` and don't need to check `apprepo.Spec.Auth != nil`

Reply:
done!